tests/Unit/RedirectsTest.php

```diff
@@ -0,0 +1,69 @@
+<?php
+namespace Automattic\LegacyRedirector\Tests\Unit;
+
+use Automattic\LegacyRedirector\Tests\Unit\MonkeyStubs;
+use WPCOM_Legacy_Redirector;
+
+/**
+ * Capability Class Unit Test
+ */
+final class RedirectsTest extends MonkeyStubs {
+
+	/**
+	 * Tests Utils::normalise_url().
+	 *
+	 * @dataProvider get_protected_redirect_data_full_url_only
+	 * @covers \Automattic\LegacyRedirector\Utils::normalise_url
+	 *
+	 * @param string $url             Full URL to parse.
+	 * @param string $expected_schema Expected return schema.
+	 * @param string $expected_domain Expected return domain.
+	 * @param string $expected_path   Expected return path.
+	 * @param string $expected_query  Expected return query.
+	 * @return void
+	 */
+	public function test_normalise_url( $url, $expected_schema, $expected_domain, $expected_path, $expected_query ) {
+
+		$expected_return = $expected_path . ( $expected_query ? '?' . $expected_query : '' );
+		$this->assertEquals( $expected_return, WPCOM_Legacy_Redirector::normalise_url( $url ) );
```

Comment:
```suggestion
		$this->assertSame( $expected_return, WPCOM_Legacy_Redirector::normalise_url( $url ) );
```

`assertSame` is a stronger check, as it also checks for type as well as value.